utilities/tools/helper/Get-PipelineStatusUrl.ps1

```diff
@@ -0,0 +1,114 @@
+﻿<#
+.SYNOPSIS
+Generate the status URL for GitHub module action workflows
+
+.DESCRIPTION
+Generate the status URL for GitHub module action workflows
+E.g.  # [![AnalysisServices: Servers](https://github.com/Azure/ResourceModules/actions/workflows/ms.analysisservices.servers.yml/badge.svg)](https://github.com/Azure/ResourceModules/actions/workflows/ms.analysisservices.servers.yml)
+
+.PARAMETER PipelineFileName
+Mandatory. The name of the workflow/pipeline file to create the badge for. For example 'platform.updateReadMe.yml'.
+Can not be provided if 'ModuleName' & 'ProviderNamespace' are provided.
+
+.PARAMETER CustomFolderPath
+Mandatory. The path to the pipeline/workflow file
+Can not be provided if 'ModuleName' & 'ProviderNamespace' are provided.
+
+.PARAMETER ModuleName
+Mandatory. The name of the module to create the url for
+Can not be provided if 'PipelineFileName' & 'CustomFolderPath' are provided.
+
+.PARAMETER ProviderNamespace
+Mandatory. The ProviderNamespace of the module to create the url for
+Can not be provided if 'PipelineFileName' & 'CustomFolderPath' are provided.
```

Comment:
I'd argue - why. Because the alternative is that I duplicate the function which is why I introduced the parameter sets in the first place. The module-based generation won't work for the Platform pipelines.

![image](https://user-images.githubusercontent.com/5365358/214306079-603d725c-6956-40fe-a3e8-203a4c3131ba.png)
![image](https://user-images.githubusercontent.com/5365358/214306199-373840f3-2b54-4121-ba38-1bc2764d0d41.png)

Maybe its easiest to have a chat.

Reply:
I wouldn't create two different functions, I see PipelineFileName and CustomFolderPath are just built from ModuleName and ProviderNameSpace, so the same logic could be used outside of the function and the same can be used for both module and platform pipelines. Anyway I agree, let's take it offline :)

Reply:
done & tested ADO & GH